src/IssueViz.Security/OpenInIDE/Api/OpenInIDERequestHandler.cs

```diff
@@ -0,0 +1,70 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.IssueVisualization.Security.OpenInIDE.Contract;
+using SonarQube.Client;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security.OpenInIDE.Api
+{
+    [Export(typeof(IOpenInIDERequestHandler))]
+    internal class OpenInIDERequestHandler : IOpenInIDERequestHandler
+    {
+        private readonly IOpenInIDEStateValidator ideStateValidator;
+        private readonly ISonarQubeService sonarQubeService;
+        private readonly ILogger logger;
+
+        [ImportingConstructor]
+        public OpenInIDERequestHandler(IOpenInIDEStateValidator ideStateValidator, ISonarQubeService sonarQubeService, ILogger logger)
+        {
+            this.ideStateValidator = ideStateValidator ?? throw new ArgumentNullException(nameof(ideStateValidator));
+            this.sonarQubeService = sonarQubeService ?? throw new ArgumentNullException(nameof(sonarQubeService));
+            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        IStatusResponse IOpenInIDERequestHandler.GetStatus()
+        {
+            throw new System.NotImplementedException();
+        }
+        
+        void IOpenInIDERequestHandler.ShowHotspot(IShowHotspotRequest request)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException(nameof(request));
+            }
+
+            // TODO: change IShowHostpotRequest server property to be a URI
```

Comment:
I'll change the property to be a URI and the IOpenInIDERequestHandler methods to be Task<X> in a separate PR.